templates/pkg/crd_sdk.go.tpl

```diff
@@ -369,3 +370,66 @@ func (rm *resourceManager) setStatusDefaults (
 		ko.Status.Conditions = []*ackv1alpha1.Condition{}
 	}
 }
+
+// updateConditions returns updated resource, true; if conditions were updated
+// else it returns nil, false
+func (rm *resourceManager) updateConditions (
+	r *resource,
+	err error,
+) (*resource, bool) {
+	ko := r.ko.DeepCopy()
+	rm.setStatusDefaults(ko)
+
+	// Terminal condition
+	var terminalCondition *ackv1alpha1.Condition = nil
+	for _, condition := range ko.Status.Conditions {
+		if condition.Type == ackv1alpha1.ConditionTypeTerminal {
+			terminalCondition = condition
+			break
+		}
+	}
+	if terminalCondition == nil {
+		terminalCondition = &ackv1alpha1.Condition{
```

Comment:
Why are we creating a terminal condition if there isn't one?

Reply:
Updated code template to not add condition when not required.